src/main/kotlin/sk/streetofcode/webapi/service/PostCommentServiceImpl.kt

```diff
@@ -0,0 +1,107 @@
+package sk.streetofcode.webapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.core.env.Environment
+import org.springframework.stereotype.Service
+import sk.streetofcode.webapi.api.EmailService
+import sk.streetofcode.webapi.api.PostCommentService
+import sk.streetofcode.webapi.api.SocUserService
+import sk.streetofcode.webapi.api.dto.PostCommentDto
+import sk.streetofcode.webapi.api.exception.AuthorizationException
+import sk.streetofcode.webapi.api.exception.BadRequestException
+import sk.streetofcode.webapi.api.exception.InternalErrorException
+import sk.streetofcode.webapi.api.exception.ResourceNotFoundException
+import sk.streetofcode.webapi.api.request.PostCommentAddRequest
+import sk.streetofcode.webapi.api.request.PostCommentEditRequest
+import sk.streetofcode.webapi.db.repository.PostCommentRepository
+import sk.streetofcode.webapi.model.PostComment
+import sk.streetofcode.webapi.model.toPostCommentDto
+import java.time.OffsetDateTime
+
+@Service
+class PostCommentServiceImpl(
+    private val postCommentRepository: PostCommentRepository,
+    private val authenticationService: AuthenticationService,
+    private val socUserService: SocUserService,
+    private val emailService: EmailService,
+    val env: Environment
+) : PostCommentService {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(PostCommentServiceImpl::class.java)
+    }
+
+    override fun getAll(postId: String): List<PostCommentDto> {
+        return postCommentRepository.findAllByPostId(postId).map { it.toPostCommentDto() }
+    }
+
+    override fun add(userId: String?, postId: String, addRequest: PostCommentAddRequest): PostCommentDto {
+        try {
+            val postComment = postCommentRepository
+                .save(
+                    PostComment(
+                        if (userId != null) socUserService.get(authenticationService.getUserId()) else null,
+                        postId,
+                        addRequest.postTitle,
+                        addRequest.commentText
+                    )
+                )
+
+            if (env.activeProfiles.contains("prod")) {
```

Comment:
Didn't we ever discuss that there should be some prop like `email_service_enabled`? Or something like that?

Anyways - I think the `env.activeProfiles.contains("prod")` check should somehow be encapsulated in `emailService` so that we don't have to think about it each time we use `emailService`. We could just move the checks into `emailService` or we could have two implementations of `EmailService` - `emailServiceProd` and `mockEmailService` or something. What do you think? ... Of course it's a detail but 🤷

Reply:
I have created prop as you suggested